slick/src/main/scala/slick/jdbc/PostgresProfile.scala

```diff
@@ -65,12 +65,14 @@ trait PostgresProfile extends JdbcProfile {
     override def createColumnBuilder(tableBuilder: TableBuilder, meta: MColumn): ColumnBuilder = new ColumnBuilder(tableBuilder, meta) {
       val VarCharPattern = "^'(.*)'::character varying$".r
       val IntPattern = "^\\((-?[0-9]*)\\)$".r
+      val DoublePattern = "[-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?".r
       override def default = meta.columnDef.map((_,tpe)).collect{
         case ("true","Boolean")  => Some(Some(true))
         case ("false","Boolean") => Some(Some(false))
         case (VarCharPattern(str),"String") => Some(Some(str))
         case (IntPattern(v),"Int") => Some(Some(v.toInt))
         case (IntPattern(v),"Long") => Some(Some(v.toLong))
+        case (DoublePattern(v), "Double") => Some(Some(v.toDouble))
```

Comment:
Yet hear you use the DoublePattern's unapply to fetch get input matching to the capture group. This means you are calling `toDouble` on a string that only consists of the exponentialPart (e.g. "e-23")